lib/src/modules/gates.dart

```diff
@@ -513,3 +513,64 @@ class Mux extends Module with InlineSystemVerilog {
     return '$control ? $d1 : $d0';
   }
 }
+
+/// A generic two-input bit index gate [Module].
```

Comment:
This isn't really a "generic" gate anymore, it's a gate specifically for returning a single bit based on an index.  Let's clarify this doc comment so that just from reading it we can tell what this module does.